### PR TITLE
feat: add entrypoint

### PR DESCRIPTION
**Pull request by laurentsenta**
This pull request replaces the old entrypoint.sh script with a new one written in Go.

We expect the conformance testing to get more complex and involve optional parameters to enable/disable parts of the test suite (like subdomains), go will be more maintainable here.

We also added the necessary flags to run the new test suite in CI.

### CODE REVIEW

**Comment by laurentsenta** [comment]
Run with gateway option:
https://github.com/ipfs/gateway-conformance/actions/runs/4428753928/jobs/7768282100

**Review by galargh — COMMENTED** [review]
Shared a few thoughts inline. I want to play with code a bit to see if clearer ideas emerge.

File: .github/actions/test/action.yml [hunk 1]
```diff
@@ -1,39 +1,55 @@
...
-    description: 'The path where the HTML report will be saved'
+    description: "The path where the HTML report will be saved"
     required: false
   markdown:
-    description: 'The path where the Markdown report will be saved'
+    description: "The path where the Markdown report will be saved"
+    required: false
+  is-subdomain:
+    description: "Enable the test suite for subdomain gateways"
     required: false
```
[the remark this answers is not in the document]
Comment (galargh) [reply .github/actions/test/action.yml:23]: I think what I want is to just be able to pass `-run` param to go test. But also for a negative version of a param to exist. If only Go regex supported negative lookahead that would be done out-of-the-box but sadly it's not.

**Comment by galargh** [comment]
Here's what came out of me playing with the entrypoint: https://github.com/ipfs/gateway-conformance/pull/15 I know the diff is awful. Let's just get together tomorrow and talk it through. I think it's going to be easier that way.

**Pull request merged**